packages/studioCMS/src/integrations/studioCMSDashboard/routes/dashboard/components/CoreVitalPanelCard.astro

```diff
@@ -0,0 +1,278 @@
+---
+import { Card, Icon, ProgressBar } from "astrolace:components";
+import type { WebVitalsResponseItem } from "studiocms-dashboard:web-vitals";
+
+interface Props {
+    lcpData: WebVitalsResponseItem[];
+    clsData: WebVitalsResponseItem[];
+    inpData: WebVitalsResponseItem[];
+}
+
+const { lcpData, clsData, inpData } = Astro.props;
+
+function msToSeconds(ms: number): number {
+    return ms / 1000;
+}
+
+function calculateClsAverage(clsValues: number[]): number {
+
+    const sum = clsValues.reduce((acc, curr) => acc + curr, 0);
+    const average = sum / clsValues.length;
+    
+    // Round average to two decimal places
+    return Math.round(average * 100) / 100;
```

Comment:
Unless we really care about precision, might just be worth using a `toFixed(2)` here

Reply:
its calculating based on the web.dev spec... :sweat_smile:

Reply:
Link? If it's a specific calculation, it's worth leaving a link to it in a comment in that case!

Reply:
https://web.dev/articles/cls#measure-cls this, plus based on help from Chris from #studio channel on Discord